src/prog_models/prognostics_model.py

```diff
@@ -1085,4 +1087,344 @@ def optimization_fcn(params):
 
         # Reset noise
         self.parameters['measurement_noise'] = m_noise
-        self.parameters['process_noise'] = p_noise                
+        self.parameters['process_noise'] = p_noise   
+
+    def generate_surrogate_dmd(self, load_functions, first_output = None, threshold_keys = None, **kwargs):
+        """
+        Generate a surrogate model to approximate the higher-fidelity model 
+
+        Parameters
+        ----------
+        load_functions : list of callable functions
+            Each index is a callable loading function of (t) -> z used to predict future loading (output) at a given time (t)
+
+        Keyword Arguments
+        -----------------
+        Includes all keyword arguments from simulate_to_threshold (except save_pts), and the following additional keywords: 
+
+        dt : Number or function, optional
+            Same as in simulate_to_threshold; for DMD, this value is the time step of the training data\n
+        save_freq : Number, optional
+            Same as in simulate_to_threshold; for DMD, this value is the time step with which the surrogate model is generated  \n
+        data_len: int, optional
+            Value between 0 and 1 that determines fraction of data resulting from simulate_to_threshold that is used to train DMD surrogate model
+            e.g. if data_len = 0.7 and the simulated data spans from t=0 to t=100, the surrogate model is trained on the data from t=0 to t=70 \n        
+        states_dmd: list, optional
```

Comment:
That sounds good. Will that cause issues anywhere else since "states" and "inputs" are used elsewhere? I'm thinking not, since this is specific to the keyword arguments, but just want to check

Reply:
It shouldn't be a problem, since they're arguments in kwargs